src/main/java/io/github/bonigarcia/wdm/WebDriverManager.java

```diff
@@ -485,15 +487,14 @@ protected File postDownload(File archive) {
 
     protected String getCurrentVersion(URL url, String driverName) {
         String currentVersion = "";
-        try {
-            currentVersion = url.getFile().substring(
-                    url.getFile().indexOf(SLASH) + 1,
-                    url.getFile().lastIndexOf(SLASH));
-        } catch (StringIndexOutOfBoundsException e) {
-            log.trace("Exception getting version of URL {} ({})", url,
-                    e.getMessage());
+        String pattern="/([^/]*?)/[^/]*?" + driverName;
+        Pattern r = Pattern.compile(pattern, Pattern.CASE_INSENSITIVE);
+        Matcher m = r.matcher(url.getFile());
+        if (m.find()) {
+            currentVersion = m.group(1);
+        }else{
+            log.trace("Exception getting version of URL {}", url);
```

Comment:
Now there are no exceptions and maybe it is better to re-word log message. Maybe to say that the URL does not contain a specified patter?

Reply:
Iog message doesn't need to be too professional, just a reminder to check the url.